### PR TITLE
Add virtualenv to ansible directory

### PR DESCRIPTION
**Pull request by LucaLanziani**
In my Linux distribution the default python is python3, thus
I was then having issues with the python scripts in the repo
where the shabang is `#!/usr/bin/env python`.

First idea was to change the shabang to `... python2` but I
realized that a virtual env can also help to pin the versions of
ansible and libraries eg. Jinja2 and make everyone env consistent.

### CODE REVIEW

File: ansible/Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+endif
+
+UPDATE_ENV: requirements.txt
+	$(virtualenv_path)/bin/pip install --upgrade setuptools
+	$(virtualenv_path)/bin/pip install --upgrade pip
+	$(virtualenv_path)/bin/pip install -r requirements.txt
+
+FLAKE8: CHECK_ENV
+	flake8
+
```
Comment (maclover7) [inline ansible/Makefile]: ✂️

File: ansible/Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+.PHONY: ENV
+
+virtualenv_path = ./virtualenv
+
+ENV: Makefile
+	virtualenv2 --distribute $(virtualenv_path)
+
+CHECK_ENV:
+ifndef VIRTUAL_ENV
+	$(error PLEASE ENTER THE VIRTUALENV BEFORE FUN THE COMMAND)
```
Comment (maclover7) [inline ansible/Makefile]: `FUN`?

File: ansible/README.md [hunk 3]
```diff
@@ -5,14 +5,27 @@
...
-   * In most cases, using pip: `pip install ansible`.
-   * If you use brew, then `brew install python2 ansible`, and then run
-   `export PYTHONPATH=$(pip2 show pyyaml | grep Location | awk '{print $2}') `
-   before you use `ansible-playbook`.
-2. Read this document.
-3. For SSH access, see the [SSH guide](../doc/ssh.md).
+1. Read this document.
+2. For SSH access, see the [SSH guide](../doc/ssh.md).
 
+### Install ansible system wide
```
Comment (maclover7) [inline ansible/README.md]: `Install Ansible systemwide`

File: ansible/README.md [hunk 4]
```diff
@@ -5,14 +5,27 @@
...
+
+Follow the [instructions to install the latest version of Ansible][ansible-install].
+* In most cases, using pip: `pip install ansible`.
+* If you use brew, then `brew install python2 ansible`, and then run
+`export PYTHONPATH=$(pip2 show pyyaml | grep Location | awk '{print $2}') `
+before you use `ansible-playbook`.
+
+### Install ansible in virtualenv
```
Comment (maclover7) [inline ansible/README.md]: Install Ansible via `virtualenv`

File: ansible/README.md [hunk 5]
```diff
@@ -5,14 +5,27 @@
...
-   before you use `ansible-playbook`.
-2. Read this document.
-3. For SSH access, see the [SSH guide](../doc/ssh.md).
+1. Read this document.
+2. For SSH access, see the [SSH guide](../doc/ssh.md).
 
+### Install ansible system wide
+
+Follow the [instructions to install the latest version of Ansible][ansible-install].
+* In most cases, using pip: `pip install ansible`.
```
Comment (maclover7) [inline ansible/README.md]: using --> use

File: ansible/Makefile (hunk 2 above)
Comment (LucaLanziani) [reply ansible/Makefile]: That was `RUNNING`, no idea why it become `FUN`, I'll fix it

**Comment by LucaLanziani** [comment]
@maclover7 comments addressed

**Comment by mhdawson** [comment]
Sorry if this is a dumb questions, but I assume that virtualenv support is consistent across platforms and available on all linux distros?

**Comment by refack** [comment]
> Sorry if this is a dumb questions, but I assume that virtualenv support is consistent across platforms and available on all linux distros?

Yes, and so is `pipenv`, both are pure python & a little bit of POSIX shell (they are kind of the equivalent to `npm`).
But since this is only required for the `ansible` host, it anyway should not affect the way we manage the "workers".

**Comment by maclover7** [comment]
@refack What are the +/- of pipenv versus virtualenv (personally I'm not familiar with it) Would you be willing to file a pull request that implements pipenv, and then we can compare both options side by side?

**Comment by refack** [comment]
> What are the +/- of pipenv versus virtualenv (personally I'm not familiar with it) Would you be willing to file a pull request that implements pipenv, and then we can compare both options side by side?

Will do.

tl;dr:
* `pip` is probably the most popular package manager for python (but defaults to global scope)
* `virtualenv` is the conventional way to enable local scoped dependencies (a la  `node_modules`)
* `pipenv` is kind of new (~1 yo) and is the latest "blessed" incarnation of a wrapper for the last two. It's the most similar to `npm` in that by default it installs dependencies in local scope.

**Comment by refack** [comment]
See seggestions in https://github.com/nodejs/build/pull/1399

**Comment by LucaLanziani** [comment]
Sorry I've should have asked before proceeding with the implementation. 

The only reason I didn't go with pipenv is that imho it adds extra features like:

- dev dependencies (do we have a dev/prod env definition? genuine question)
- lock file (do we need this on releases? genuine question)
- install packages and update requirements for you (for this I can see a use)

over something that works already well enough on simple cases like the one we have here (I might be underestimating the complexity of the repo).

That said, more than happy to use pipenv if you guys thinks it's better/we need this functionalities.

**Comment by refack** [comment]
> Sorry I've should have asked before proceeding with the implementation.

No worries. I mostly like it because it's newer and is more `npm` like. Only two features it has that make a difference are newer [PEP508](https://www.python.org/dev/peps/pep-0508/) dependency URIs, and it captures the required python engined needed.

What I really wanted to do is get the code to be 2/3 compatible (https://github.com/nodejs/build/pull/1399/commits/2b5e0639855db44d5d1fbf96d7e244dc1930f4e0).
With a formal python2 EOL date, we need to start moving...

**Comment by sam-github** [comment]
stale, and not what we are doing ATM.
